### DISCUSSION TITLE
App hosted on Amplify SSL error

### DISCUSSION THREAD

**Discussion by rajatdhoot123** [planetscale/discussion#348] [Q&A] [answered]
My app is hosted on aws amplify ssl is issued by aws certificate manager don't know how to verify it.

**Comment by dbussink** [comment]
Can you please provide more information? What language / framework(s) are you using? What is the exact error message that you see (please include it in full and don't truncate / remove parts of it). 

Also the TLS certificate issues by AWS certificate managed should not have anything to do with the TLS certificate that PlanetScale presents.

**Comment by rajatdhoot123** [comment, marked as answer]
My Project is based on next js which is hosted on amazon amplify when I tried with the connection url 
`DATABASE_URL=mysql://xxxxxxxxxxxxxxxxxx:xxxxxxxxxxxxxxxxxxx@ap-south.connect.psdb.cloud/xxxxxxxx?ssl={"rejectUnauthorized":true}` it will through SSL error which is working fine on my localhost

**Reply by dbussink** [reply]
> My Project is based on next js which is hosted on amazon amplify when I tried with the connection url
> `DATABASE_URL=mysql://xxxxxxxxxxxxxxxxxx:xxxxxxxxxxxxxxxxxxx@ap-south.connect.psdb.cloud/xxxxxxxx?ssl={"rejectUnauthorized":true}` it will through SSL error which is working fine on my localhost

Can you please show the actual error that you get with all the details included? If it works on localhost though, I'm not sure why anything would be different on Amazon.

**Reply by rajatdhoot123** [reply]
<img width="710" alt="image" src="https://user-images.githubusercontent.com/18528826/204588722-f2f776e0-ea9d-44c8-b411-2b86ca5130b0.png">
This is the exact error
`TypeError: Unknown SSL profile '
{
    "rejectUnauthorized": true
}`

**Reply by dbussink** [reply]
> This is the exact error `TypeError: Unknown SSL profile ' { "rejectUnauthorized": true }`

Can you also provide additional information like the specific driver you're using? You mentioned using Next.js, but how are you connecting to PlanetScale? 

Are you using perhaps `mysql2`? I can find https://github.com/sidorares/node-mysql2/blob/e5727f9447567aab31c379bfc4ed351d9ac58cb5/lib/connection_config.js#L246 which shows this style of error message? 

Looking at https://github.com/sidorares/node-mysql2/blob/e5727f9447567aab31c379bfc4ed351d9ac58cb5/lib/connection_config.js#L135 the problem is probably that it somehow parses the value of `ssl=` as a string, not as a hash so it tries to load a profile name. This means there's something probably happening with your environment variable loading that parses things wrong. 

How are you loading that environment variable in your application and what does the logic look like that creates the database connection?

**Reply by rajatdhoot123** [reply]
Yes, the issue is improper parsing of environment variables.

Thanks, appreciate digging into libraries for finding the possible issue and help the community.
Excellent community support.
